vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -179,7 +179,11 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 	// DEPRECATED: remove and replace with something more accurate
 	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
 
-	flags.StringVar(&loadingRules.ExplicitPath, "kubeconfig", "", "Path to the kubeconfig file to use for CLI requests.")
+	if clientcmd.UseOpenShiftKubeConfigValues {
+		flags.StringVar(&loadingRules.ExplicitPath, clientcmd.OpenShiftKubeConfigFlagName, "", "Path to the kubeconfig file to use for CLI requests.")
```

Comment:
While you're at it, since as the followup I'll revive the kubeconfig PR I have, maybe we can just squash both changes so that `kubeconfig` is present always, we'll need it either way.